### DISCUSSION TITLE
Add file access for hdf5 files

### DISCUSSION THREAD

**Discussion by alackpour** [mlcommons/croissant#637] [Ideas]
Idea: add file access for hdf5 files.

Are any developers and/or users interested in this feature?
(I saw that the proposed geo-croissant spec includes hdf5 file access: https://www.earthdata.nasa.gov/learn/blog/introducing-croissant-format-machine-learning-datasets)

Does anyone else want to access hdf5 files using croissant?

**Comment by benjelloun** [comment]
Sounds like a good idea to me! 

The simplest integration I can think of is supporting the definitions of Croissant FileObject and Filesets that can read inside HDF5 files and groups, similarly to how we support zip / tar archives today.

Is this what you had in mind?

**Reply by alackpour** [reply]
Hi, yes, that's exactly what I was proposing.
I expect that both data and metadata could be stored inside one or more h5 files.
Is h5 file access part of your existing roadmap for the project?

**Reply by stefanches7** [reply]
hdf5 and Zarr too would be very useful for the *Bioimaging* community too!
